skyportal/handlers/api/comment.py

```diff
@@ -87,15 +87,56 @@ def post(self):
         obj_id = data['obj_id']
         # Ensure user/token has access to parent source
         _ = Source.get_obj_if_owned_by(obj_id, self.current_user)
-        user_group_ids = [g.id for g in self.current_user.groups]
         user_accessible_group_ids = [g.id for g in self.current_user.accessible_groups]
-        group_ids = data.pop("group_ids", user_group_ids)
-        group_ids = [gid for gid in group_ids if gid in user_accessible_group_ids]
+        user_accessible_filter_ids = [
+            filtr.id
+            for g in self.current_user.accessible_groups
+            for filtr in g.filters
+            if g.filters is not None
+        ]
+        group_ids = data.pop("group_ids", user_accessible_group_ids)
+        group_ids = [
+            int(gid) for gid in group_ids if int(gid) in user_accessible_group_ids
+        ]
         if not group_ids:
             return self.error(
                 f"Invalid group IDs field ({group_ids}): "
                 "You must provide one or more valid group IDs."
             )
+
+        # Only post to groups source/candidate is actually associated with
+        if DBSession().query(Candidate).filter(Candidate.obj_id == obj_id).all():
+            candidate_group_ids = [
+                f.group_id
+                for f in (
+                    DBSession()
```

Comment:
Perhaps splitting these out.  I.e., assign a variable to the inner query, then operate on that, etc.

That way, there are some human-readable handles to what those queries bring in.